agent/ui_endpoint.go

```diff
@@ -117,7 +119,22 @@ RPC:
 	if out.Dump == nil {
 		out.Dump = make(structs.NodeDump, 0)
 	}
-	return out.Dump, nil
+
+	// Use empty list instead of nil
+	for _, info := range out.ImportedDump {
+		if info.Services == nil {
+			info.Services = make([]*structs.NodeService, 0)
+		}
+		if info.Checks == nil {
+			info.Checks = make([]*structs.HealthCheck, 0)
+		}
+	}
+
+	if out.ImportedDump == nil {
+		out.ImportedDump = make(structs.NodeDump, 0)
+	}
```

Comment:
nit: Not necessary to allocate since we `append` to a non-nil `out.Dump` anyways (ImportedDump is optional)

Reply:
I believe that for the UI they prefer that nothing be nil. Presumably it's javascript related.

Reply:
Should check with John before changing this

Reply:
This endpoint joins `out.Dump` and `out.ImportedDump` into one though so it shouldn't matter, right?

Reply:
Ah yep good catch